code/modules/mining/equipment_locker.dm

```diff
@@ -533,41 +620,70 @@
 			triggered = TRUE
 
 	if(triggered)
-		usr.visible_message("<span class='warning'>The [src] overloads and activates!</span>")
+		usr.visible_message("<span class='warning'>[src] overloads and activates!</span>")
 		feedback_add_details("jaunter","E") // EMP accidental activation
 		activate(usr)
 
-/obj/item/device/wormhole_jaunter/proc/chasm_react(mob/user)
+/obj/item/device/wormhole_lifebelt/chasm_react(mob/user)
 	if(user.get_item_by_slot(slot_belt) == src)
-		user << "Your [src] activates, saving you from the chasm!</span>"
 		feedback_add_details("jaunter","C") // chasm automatic activation
-		activate(user)
+		. = activate(user)
+		if(.)
+			user << "[src] activates, saving you from the chasm!</span>"
 	else
-		user << "The [src] is not attached to your belt, preventing it from saving you from the chasm. RIP.</span>"
+		user << "[src] is not attached to your belt, preventing it from saving you from the chasm. RIP.</span>"
+		. = FALSE
 
 
 /obj/effect/portal/wormhole/jaunt_tunnel
 	name = "jaunt tunnel"
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "bhole3"
-	desc = "A stable hole in the universe made by a wormhole jaunter. Turbulent doesn't even begin to describe how rough passage through one of these is, but at least it will always get you somewhere near a beacon."
+
+	desc = "A stable hole in the universe made by a wormhole lifebelt. \
+		Turbulent doesn't even begin to describe how rough passage through \
+		one of these is, but at least it will always get you somewhere near a \
+		beacon."
 
 /obj/effect/portal/wormhole/jaunt_tunnel/teleport(atom/movable/M)
 	if(istype(M, /obj/effect))
 		return
 	if(istype(M, /atom/movable))
 		if(do_teleport(M, target, 6))
 			// KERPLUNK
+			M.visible_message("<span class='warning'>[M] appears from nowhere!</span>", null, "<span class='warning'>There is a loud ker-plunk noise!</span>")
 			playsound(M,'sound/weapons/resonator_blast.ogg',50,1)
+
 			if(iscarbon(M))
 				var/mob/living/carbon/L = M
+				// Golems are extremely resistant to the stress of
+				//  wormhole travel.
+				if(isgolem(L))
+					L.Weaken(1)
+					return
 				L.Weaken(3)
 				if(ishuman(L))
 					shake_camera(L, 20, 1)
+					// aproximately x3 telearmor rads
+					L.rad_act(50)
 					spawn(20)
 						if(L)
 							L.vomit(20)
 
+/proc/isgolem(mob/M)
+	if(istype(M, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		if(H.dna && istype(H.dna.species, /datum/species/golem))
+			return TRUE
+	return FALSE
+
+/obj/item/device/wormhole_lifebelt/advanced
```

Comment:
NO ESCAPE. But seriously, it costs like 2000 now, we can still disable it if you hate it that much.

My PR also made free golems able to buy jaunters but not use them to escape to the station, which is good.